### PR TITLE
[WIP] add pipeline-level nf-test

### LINKED ISSUE #61: Implement nf-test
### Description of feature

I'm going to switch the github ci to pytest-workflow so that github can monitor the consistency of pipeline output instead of just checking that PRs don't make the pipeline fail

### PR DESCRIPTION
**Pull request by pinin4fjords**
Close https://github.com/nf-core/differentialabundance/pull/61

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by pinin4fjords** [comment]
@nf-core-bot fix linting

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,9 +76,27 @@ jobs:
         with:
           version: "${{ matrix.NXF_VER }}"
 
-      - name: Run pipeline with test data
-        # You can customise CI pipeline run tests as required
-        # For example: adding multiple test runs with different parameters
-        # Remember that you can parallelise this by using strategy.matrix
+      - name: Cache nf-test installation
+        id: cache-software
```
Comment (Emiller88) [inline .github/workflows/ci.yml:80]: I think this needs a hash or something.

File: tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+        when {
+            params {
+                outdir     = "results"
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success }
```
Comment (Emiller88) [inline tests/main.nf.test:18]: This is bare minimum. Added a snapshot with some of the results would be a great follow up!

File: tests/nextflow.config
Comment (Emiller88) [inline tests/nextflow.config:1]: You can remove this file I think 🤔

**Review by edmundmiller — APPROVED** [review]
Once this merges I'll make a PR with https://github.com/nf-core/methylseq/pull/350

File: tests/main.nf.test (hunk 2 above)
Comment (pinin4fjords) [reply tests/main.nf.test:18]: This is the pipeline level, i.e. there are no emits. What would be the way to snapshot?

Comment (maxulysse) [reply tests/main.nf.test:18]: It is indeed the bare minimum, but I'm happy with that already to be honest

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -16,22 +16,57 @@ concurrency:
...
+        NXF_VER: ${{ fromJson(needs.define_nxf_versions.outputs.matrix) }}
+        tags: ["${{ fromJson(needs.changes.outputs.tags) }}"]
+        compute_profile:
+          - "docker"
+        test_profile:
+          - test
+          - test_nogtf
+          - test_affy
+          - test_soft
+          - test_maxquant
```
Comment (maxulysse) [inline .github/workflows/ci.yml:68]: oh I see what you're doing there.
I'd rather have that done with different tests and snapshots, similarly to how we do with the multiple workflow tests in fetchngs

**Comment by pinin4fjords** [comment]
I'm actually going to close this, things have moved on in standards for pipeline-level testing, and we need to start again.

**Pull request closed without merging**